react/src/components/HeaderNavBar/HeaderNavBar.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from 'react';
+import { Layout } from 'antd';
+import useAuthenticatedUser from '@hazmapper/hooks/user/useAuthenticatedUser';
+import { useNavigate } from 'react-router-dom';
+import { Button, InlineMessage, LoadingSpinner } from '@tacc/core-components';
+import styles from './HeaderNavBar.module.css';
+
+export const HeaderNavBar: React.FC = () => {
+  const { Header } = Layout;
+  const navigate = useNavigate();
+
+  const {
+    data: userData,
+    isLoading: isUserLoading,
+    error: isUserError,
+    refetch,
+  } = useAuthenticatedUser();
+
+  const handleLogin = () => {
+    const url = `/login?to=${encodeURIComponent(location.pathname)}`;
+    navigate(url);
+  };
+
+  useEffect(() => {
+    if (userData) refetch();
+  }, [userData, refetch]);
```

Comment:
can the use of refetch be dropped?  i tested locally with viewing a public map and then logged in and all worked well without it.